Mage.Sets/src/mage/cards/s/ShanidSleepersScourge.java

```diff
@@ -0,0 +1,92 @@
+import mage.abilities.effects.common.continuous.GainAbilityAllEffect;
+import mage.abilities.effects.common.continuous.GainAbilityControlledEffect;
+import mage.constants.*;
+import mage.abilities.keyword.MenaceAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.filter.FilterPermanent;
+import mage.filter.FilterSpell;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.common.FilterLandCard;
+import mage.game.Game;
+import mage.game.events.DrawCardEvent;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+
+/**
+ * @author freaisdead
+ */
+public final class ShanidSleepersScourge extends CardImpl {
+
+    private static final FilterCreaturePermanent otherLegendaryCreaturesFilter = new FilterCreaturePermanent("other legendary creatures");
+    private static final FilterSpell legendarySpellFilter = new FilterSpell("a legendary spell");
+    private static final FilterPermanent legendaryLandFilter = new FilterPermanent("a legendary land");
+
+    static {
+        otherLegendaryCreaturesFilter.add(SuperType.LEGENDARY.getPredicate());
+
+        legendarySpellFilter.add(SuperType.LEGENDARY.getPredicate());
+
+        legendaryLandFilter.add(CardType.LAND.getPredicate());
+        legendaryLandFilter.add(SuperType.LEGENDARY.getPredicate());
+    }
+
+    public ShanidSleepersScourge(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{R}{W}{B}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.KNIGHT);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(4);
+
+        // Menace
+        this.addAbility(new MenaceAbility());
+
+        // Other legendary creatures you control have menace.
+        this.addAbility(new SimpleStaticAbility(new GainAbilityControlledEffect(
+                new MenaceAbility(),
+                Duration.WhileOnBattlefield,
+                otherLegendaryCreaturesFilter,
+                true)));
+        // Whenever you play a legendary land or cast a legendary spell, you draw a card and you lose 1 life.
+        Ability castLegendarySpellAbility = new SpellCastControllerTriggeredAbility(
+                new DrawCardSourceControllerEffect(1),
+                legendarySpellFilter,
+                false);
+        castLegendarySpellAbility.addEffect(new LoseLifeSourceControllerEffect(1));
+        this.addAbility(castLegendarySpellAbility);
+        Ability playLegendaryLandAbility = new EntersBattlefieldControlledTriggeredAbility(
+                Zone.BATTLEFIELD,
+                new DrawCardSourceControllerEffect(1),
+                legendaryLandFilter,
+                false);
+        playLegendaryLandAbility.addEffect(new LoseLifeSourceControllerEffect(1));
+        this.addAbility(playLegendaryLandAbility);
```

Comment:
If you open up the GUI, does the generate text for these two abilities work out correctly? I expect that you would get two sentences, one for each ability.

Reply:
I will check, but why build failed?

Reply:
it looks like it’s due to something unrelated

Reply:
@Alex-Vasile yes... It's two separate sentences... :( how can I make them a single effect without doing crazy stuff ?

Reply:
This should technically get a new effect, something like `DrawAndLoseLife` since:
1. There are a great number of cards that do something like "Draw X cards and you lose X life"
2. These effects should technically implemented as a single effect. Having it as multiple effects means that they technically do not happen at the same time (first you draw and then you lose the life) but they should occur at the same time. See https://github.com/magefree/mage/pull/9304#discussion_r927718153. For a similar bug (but in reverse, i.e. the effects were done at the same time rather than sequentially) see the fixed "Balancing Act" issue in #6761.

Reply:
made changes, but test is not working anymore... seems like the ability is not triggered